app/components/work_package_relations_tab/relations_mediator.rb

```diff
@@ -29,9 +29,21 @@
 #++
 
 class WorkPackageRelationsTab::RelationsMediator
-  RelationGroup = Data.define(:type, :visible_relations, :ghost_relations) do
-    def initialize(type:, visible_relations:, ghost_relations:)
-      super(type: ActiveSupport::StringInquirer.new(type), visible_relations:, ghost_relations:)
+  RelationGroup = Data.define(:type, :work_package, :visible_relations, :ghost_relations, :closest_relation) do
+    def initialize(type:, work_package:, visible_relations:, ghost_relations:)
+      type = ActiveSupport::StringInquirer.new(type.to_s)
+      closest_relation = self.class.closest_relation(type, visible_relations + ghost_relations)
```

Comment:
👍🏻 good that we're only calculating the closest relation once on instantiation (that was my mistake).

We could remove the type param from `self.closest_relation` and check the type here instead.

```suggestion
      closest_relation = self.class.closest_relation(visible_relations + ghost_relations) if type.follows?
```